pulp_rpm/app/migrations/0047_modulemd_datefield.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 3.2.16 on 2022-11-08 19:12
+
+from django.db import migrations, models
+
+import yaml
+
+
+def parse_date(apps, schema_editor):
+    """Parse date from snippet."""
+    ModulemdObsolete = apps.get_model("rpm", "ModulemdObsolete")
+
+    obsoletes_to_update = []
+
+    for obsolete in ModulemdObsolete.objects.all():
+        parsed_snippet = yaml.safe_load(obsolete.snippet)
+        obsolete.modified = parsed_snippet['data']['modified']
+        obsoletes_to_update.append(obsolete)
+
+    ModulemdObsolete.objects.bulk_update(obsoletes_to_update, ["modified"])
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('rpm', '0046_rbac_perms'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='modulemdobsolete',
+            name='eol_date',
+            field=models.TextField(null=True),
+        ),
+        migrations.AlterField(
+            model_name='modulemdobsolete',
+            name='modified',
+            field=models.TextField(),
+        ),
+        migrations.RunPython(parse_date),
```

Comment:
So the field type is altered in place, and the data is only changed afterwards, and that works?

Reply:
It does. Also added forgotten update for `eol_date` if exists.

```
for o in ModulemdObsolete.objects.all():
   ...:     print(o.modified)
   ...:     print(o.eol_date)
   ...: 
2022-01-24 08:54:00+00:00
2021-06-01 00:00:00+00:00
2022-03-03 15:28:00+00:00
2022-06-01 00:00:00+00:00

------
Running migrations:
  Applying rpm.0047_modulemd_datefield... OK
------

for o in ModulemdObsolete.objects.all():
   ...:     print(o.modified)
   ...:     print(o.eol_date)
   ...: 
2022-01-24T08:54Z
2021-06-01T00:00Z
2022-03-03T15:28Z
2022-06-01T00:00Z
```

Reply:
but rather updated to safe-way migrate